packages/js/product-editor/src/blocks/schedule-sale/edit.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * External dependencies
+ */
+import { DateTimePickerControl } from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+import { useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { ToggleControl } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+import { createElement, useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import moment from 'moment';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We need this to get the datetime format for the DateTimePickerControl.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { getSettings } from '@wordpress/date';
+
+/**
+ * Internal dependencies
+ */
+import { ScheduleSalePricingBlockAttributes } from './types';
+import { useValidation } from '../../hooks/use-validation';
+
+export function Edit( {}: BlockEditProps< ScheduleSalePricingBlockAttributes > ) {
+	const blockProps = useBlockProps( {
+		className: 'wp-block-woocommerce-product-schedule-sale-fields',
+	} );
+
+	const dateTimeFormat = getSettings().formats.datetime;
+
+	const [ showScheduleSale, setShowScheduleSale ] = useState( false );
+
+	const [ salePrice ] = useEntityProp< string | null >(
+		'postType',
+		'product',
+		'sale_price'
+	);
+
+	const isSalePriceGreaterThanZero =
+		Number.parseFloat( salePrice || '0' ) > 0;
+
+	const [ dateOnSaleFromGmt, setDateOnSaleFromGmt ] = useEntityProp<
+		string | null
+	>( 'postType', 'product', 'date_on_sale_from_gmt' );
+
+	const [ dateOnSaleToGmt, setDateOnSaleToGmt ] = useEntityProp<
+		string | null
+	>( 'postType', 'product', 'date_on_sale_to_gmt' );
+
+	const today = moment().toISOString();
+
+	function handleToggleChange( value: boolean ) {
+		recordEvent( 'product_pricing_schedule_sale_toggle_click', {
+			enabled: value,
+		} );
+
+		setShowScheduleSale( value );
+
+		if ( value ) {
+			setDateOnSaleFromGmt( today );
+			setDateOnSaleToGmt( '' );
+		} else {
+			setDateOnSaleFromGmt( '' );
+			setDateOnSaleToGmt( '' );
+		}
+	}
+
+	// Hide and clean date fields if the user manually change
+	// the sale price to zero or less.
+	useEffect( () => {
+		if ( ! isSalePriceGreaterThanZero ) {
+			setShowScheduleSale( false );
+			setDateOnSaleFromGmt( '' );
+			setDateOnSaleToGmt( '' );
+		}
+	}, [ isSalePriceGreaterThanZero ] );
+
+	// Automatically show date fields if `from` or `to` dates have
+	// any value.
+	useEffect( () => {
+		if ( dateOnSaleFromGmt || dateOnSaleToGmt ) {
+			setShowScheduleSale( true );
+		}
+	}, [ dateOnSaleFromGmt, dateOnSaleToGmt ] );
+
+	const isDateOnSaleToGmtValid = useValidation(
+		'product/date_on_sale_to_gmt',
+		() =>
+			! showScheduleSale ||
+			! dateOnSaleToGmt ||
+			moment( dateOnSaleFromGmt ).isBefore( dateOnSaleToGmt, 'minute' )
```

Comment:
This seems to actually cause an issue when the times are close.  If I set the end date to be a few minutes after, it's not valid:
<img width="683" alt="Screen Shot 2023-04-05 at 10 03 14 AM" src="https://user-images.githubusercontent.com/10561050/230152863-da6ed208-c410-49e4-a5d0-c4416082f198.png">

Side note (not caused by this PR) but I can save the product even when fields are invalid.  The "Save" button is grayed out but can still be clicked and the information persists:

<img width="704" alt="Screen Shot 2023-04-05 at 10 02 50 AM" src="https://user-images.githubusercontent.com/10561050/230152994-4454d586-a210-4a7e-a19f-ceed01193b99.png">

Reply:
Yes, because the diff should be more than 1 minute.

On the other hand we should talk about this more in details later on because it does not make sense a tiny time difference for an offer to live right. What is gonna be the less difference between both dates?

cc: @jarekmorawski